packages/astro/src/vite-plugin-html/transform/utils.ts

```diff
@@ -25,5 +25,5 @@ export function needsEscape(value: any): value is string {
 	return typeof value === 'string' && (value.includes('`') || value.includes('${'));
 }
 export function escape(value: string) {
-	return value.replace(/`/g, '\\`').replace(/\$\{/g, '\\${');
+	return value.replace(/(\\*)\`/g, '\\`').replace(/\$\{/g, '\\${');
```

Comment:
This is definitely better but might fail if the backslash itself is escaped. I think we might need to use a different approach entirely? I'm going to look into some options.

Reply:
Yeah. But I have a question, is there any cases in this scenario that the backslash itself is escaped. On the other hand, before the `escape` function call, it should pass `needEscape` check.